barista-tracing/src/main/java/com/markelliot/barista/tracing/EmptyTrace.java

```diff
@@ -30,26 +30,31 @@ public final class EmptyTrace implements Trace {
 
     @Override
     public Span rootSpan(String opName) {
-        return singletonSpan;
+        return getSpan();
     }
 
     @Override
     public Span rootSpan(Supplier<String> opName) {
-        return singletonSpan;
+        return getSpan();
     }
 
     @Override
     public Span withParent(String parentId, String opName) {
-        return singletonSpan;
+        return getSpan();
     }
 
     @Override
     public Span withParent(String parentId, Supplier<String> opName) {
-        return singletonSpan;
+        return getSpan();
     }
 
     @Override
     public String traceId() {
         return traceId;
     }
+
+    private Span getSpan() {
+        Spans.setThreadSpan(singletonSpan);
```

Comment:
This should happen in the empty span constructor to match default spans.

Reply:
If we linked initializing the thread local with constructing the span then we would have to end up shifting creation of the empty span from construction of the trace to calls on methods of the trace and would have to create multiple empty spans.

While inconsistent with `DefaultTrace` this approach seemed like the simplest, lowest cast way to accomplish the outcome